### PR TITLE
Update rfhub2 to handle rfwk 4.0+

### PR DESCRIPTION
**Pull request by MaciejWiczk**
Hopefuly, this will enable RfHub2 to support RFWK 4.0+. I do not have an acess to large RFWK repository anymore, so maybe someone could test that locally, before releasing?
@MarcinMaciaszek ?

### CODE REVIEW

File: tests/unit/cli/statistics/statistics_extractor.py [hunk 1]
```diff
@@ -39,8 +51,9 @@ def test_compute_statistics_should_return_expected_data(self):
 
     def test_parse_xml_keywords_should_return_expected_statistics(self):
         result = self.statistics_extractor.parse_xml_keywords()
+        self.maxDiff = None
```
Comment (pbylicki) [inline tests/unit/cli/statistics/statistics_extractor.py]: is this needed here?

File: rfhub2/cli/statistics/statistics_extractor.py [hunk 2]
```diff
@@ -95,8 +95,16 @@ def parse_xml_keywords(self) -> List[XmlKeyword]:
             )
             for xml_keyword in xml_keywords
             if xml_keyword.attrib.get("library") is not None
+            and self.has_keyword_passed(xml_keyword)
         ]
 
+    @staticmethod
+    def has_keyword_passed(xml_keyword: XmlKeyword) -> bool:
+        """
+        Checks is keyword has a ``PASS`` status in output.xml file
```
Comment (pbylicki) [inline rfhub2/cli/statistics/statistics_extractor.py]: Checks is => Checks if

File: rfhub2/cli/statistics/statistics_extractor.py [hunk 3]
```diff
@@ -95,8 +95,16 @@ def parse_xml_keywords(self) -> List[XmlKeyword]:
             )
             for xml_keyword in xml_keywords
             if xml_keyword.attrib.get("library") is not None
+            and self.has_keyword_passed(xml_keyword)
         ]
 
+    @staticmethod
```
Comment (pbylicki) [inline rfhub2/cli/statistics/statistics_extractor.py:101]: Could you expand the docstring to include why this change is needed in 4.x and confirm whether it affects rfhub when it's run with robotframework < 4.0?

Comment (MaciejWiczk) [reply rfhub2/cli/statistics/statistics_extractor.py:101]: @pbylicki sure. This is not exactly needed by rfwk 4+, but would be a safeguard, to not include failed keywords in statistics gathering. I will highlight that in docstring

File: tests/unit/cli/statistics/statistics_extractor.py (hunk 1 above)
Comment (MaciejWiczk) [reply tests/unit/cli/statistics/statistics_extractor.py]: no, not at all

**Review by pbylicki — APPROVED** [review]
